auth-web/src/util/common-util.ts

```diff
@@ -268,4 +268,9 @@ export default class CommonUtils {
   static trimTrailingSlashURL (url) {
     return (url) ? url.trim().replace(/\/+$/, '') : ''
   }
+
+  /** use entity type to get NR type from enum */
+  static mapEntityTypeToNrType (entityTypeCd: string): string {
+    return NrRequestTypes[entityTypeCd]
```

Comment:
What happens here if entityTypeCd is undefined (which it would be the optional chaining on affiliations-factory.ts line 161 fails)?

(I'm not actually sure whether that would cause an error here. You may need a check here.)

Reply:
I just tested by manually setting it to undefined and it does not cause any errors. It basically will just show "Name Request" with no emdash or type appended.

Reply:
Awesome. It probably returns undefined, and the code in affiliations-factory check for non-falsy nrType so it's safe 👍

Reply:
I don't think it's proper to use a mapping from `entityTypeCd` to nr type. Instead, it should be from `requestTypeCd` to nr type. Because some codes like `BEC`, `BERE` etc. only exit in `requestTypeCd`. So please tests more cases like alternation, restoration, amalgamation etc. to see if it shows incorrect nr type.

Reply:
I see yeah that could be an issue. The `business`/`business.nameRequest` object returned from auth-api does not appear to have a `requestTypeCd`. Do you know where the `requestTypeCd` get's returned from?

Reply:
Yeah it's the same question I got yesterday. Please see the code in the following link. The response that fetches affiliations in fact returns `requestTypeCd`, but it's not mapped into `business.nameRequest`. And also check Network for the details.
https://github.com/bcgov/sbc-auth/blob/main/auth-web/src/store/modules/business.ts#L137

Reply:
I'm not following this conversation 100%, but I do want to point out that NRs have different codes than LEAR. There are tables in namex and Legal API to convert from the NR code to the LEAR codes. Please be clear about which codes you are trying to convert to a string (or applying logic to).

Reply:
> I see yeah that could be an issue. The `business`/`business.nameRequest` object returned from auth-api does not appear to have a `requestTypeCd`. Do you know where the `requestTypeCd` gets returned from?

I believe the NR object has a property named requestTypeCd (or request_type_cd).

Reply:
> Yeah it's the same question I got yesterday. Please see the code in the following link. The response that fetches affiliations in fact returns `requestTypeCd`, but it's not mapped into `business.nameRequest`. And also check Network for the details. https://github.com/bcgov/sbc-auth/blob/main/auth-web/src/store/modules/business.ts#L137

I see yeah checking the Network tab did reveal that. Mapping it into `business.nameRequest` worked for me!

Brief explanation Sev, when I was using `entityTypeCd`, it would return some codes from this [NrRequestTypeCodes enum](https://github.com/bcgov/bcrs-shared-components/blob/main/src/enums/name-request-codes.ts) (which could be mapped to an NR type string), as well as some codes from this [EntityType enum](https://github.com/bcgov/namerequest/blob/feature-way-of-navigating/src/enums/entity-type.ts) (which it isn't clear what NR type to map it too). Using `requestTypeCd` seems to map each Name Request to an appropriate NR type.

Reply:
I'm OK to resolve this conversation if you are.